cmd/aks-periscope/aks-periscope.go

```diff
@@ -55,35 +105,104 @@ func main() {
 	osmCollector := collector.NewOsmCollector()
 	smiCollector := collector.NewSmiCollector()
 
-	collectors := []interfaces.Collector{
-		containerLogsCollector,
-		dnsCollector,
-		kubeObjectsCollector,
-		networkOutboundCollector,
+	selectedCollectors := selectCollectors(
+		map[string]interfaces.Collector{
+			containerLogsCollector.GetName():           containerLogsCollector,
+			systemLogsCollector.GetName():              systemLogsCollector,
+			networkOutboundCollector.GetName():         networkOutboundCollector,
+			ipTablesCollector.GetName():                ipTablesCollector,
+			nodeLogsCollector.GetName():                nodeLogsCollector,
+			dnsCollector.GetName():                     dnsCollector,
+			kubeObjectsCollector.GetName():             kubeObjectsCollector,
+			kubeletCmdCollector.GetName():              kubeletCmdCollector,
+			systemPerfCollector.GetName():              systemPerfCollector,
+			helmCollector.GetName():                    helmCollector,
+			osmCollector.GetName():                     osmCollector,
+      			smiCollector.GetName():                     smiCollector,
+		})
+
+	//diagnosers
+	//NOTE currently the collector instance objects are shared between the collector itself and things which use it as a dependency
+	networkConfigDiagnoser := diagnoser.NewNetworkConfigDiagnoser(dnsCollector, kubeletCmdCollector)
+	networkOutboundDiagnoser := diagnoser.NewNetworkOutboundDiagnoser(networkOutboundCollector)
+	selectedDiagnosers := selectDiagnosers(
+		map[string]interfaces.Diagnoser{
+			networkConfigDiagnoser.GetName():   networkConfigDiagnoser,
+			networkOutboundDiagnoser.GetName(): networkOutboundDiagnoser,
+		})
+
+	//exporters
+	azureBlobExporter := exporter.NewAzureBlobExporter(creationTimeStamp, hostname)
+	selectedExporters := selectExporters(
+		map[string]interfaces.Exporter{
+			azureBlobExporter.GetName():    azureBlobExporter,
+		})
+
+	return selectedCollectors, selectedDiagnosers, selectedExporters
+}
+
+// selectCollectors select the collectors to run by looking up the value from the config
+func selectCollectors(allCollectorsByName map[string]interfaces.Collector) []interfaces.Collector {
+	collectors := []interfaces.Collector{}
+
+	enabledCollectorString, found := os.LookupEnv("ENABLED_COLLECTORS")
```

Comment:
💡 What will be associated `deployment.yaml` changes will look like? For this new var `ZIP_AND_EXPORT`

Reply:
https://github.com/davidkydd/aks-periscope/blob/0071201dd41cd159e87e5a574bdfbcc271c122b1/deployment/aks-periscope-arc.yaml#L147